src/TimeModeling/Modeling/lsrtm_objective_parallel.jl

```diff
@@ -27,3 +27,31 @@ function lsrtm_objective(model::Model, source::judiVector, dObs::judiVector, dm;
     # first value corresponds to function value, the rest to the gradient
     return obj, gradient
 end
+
+function lsrtm_objective(model::Array{Model,1}, source::Array{judiVector{T,Array{T,2}},1}, dObs::Array{judiVector{T,Array{T,2}},1}, dm::Array; options=Options(), nlind=false) where T
+# lsrtm_objective function for multiple sources and multiple vintages. The function distributes the sources and the input data amongst the available workers.
+
+    obj = Array{T, 1}(undef, length(dObs))
+    gradient = Array{Array{T,2}, 1}(undef, length(dObs))
+
+    @sync for i = 1:length(dObs)
+        @async obj[i], gradient[i] = lsrtm_objective(model[i], source[i], dObs[i], dm[i]; options=options, nlind=nlind)
+    end
+
+    # first value corresponds to function value, the rest to the gradient
+    return obj, gradient
+end
+
+function lsrtm_objective(model::Model, source::Array{judiVector{T,Array{T,2}},1}, dObs::Array{judiVector{T,Array{T,2}},1}, dm::Array; options=Options(), nlind=false) where T
+# lsrtm_objective function for multiple sources and multiple vintages. The function distributes the sources and the input data amongst the available workers.
+
+    obj = Array{T, 1}(undef, length(dObs))
+    gradient = Array{Array{T,2}, 1}(undef, length(dObs))
+
+    @sync for i = 1:length(dObs)
```

Comment:
Again you're making your life complicated and replicating code. Make `model. source, dobs,...` the input of `judipmap` and have it take care of the parallelism with dispatch.